neuvector-controller-5.3.yaml

```diff
@@ -31,16 +33,50 @@ pipeline:
       deps: golang.org/x/net@v0.23.0 golang.org/x/crypto@v0.22.0 github.com/opencontainers/image-spec@v1.1.0 github.com/docker/distribution@v2.8.2-beta.1+incompatible github.com/opencontainers/runc@v1.1.12 github.com/containerd/containerd@v1.6.26 google.golang.org/grpc@v1.56.3 google.golang.org/protobuf@v1.33.0 github.com/golang/protobuf@v1.5.4 github.com/docker/docker@v26.1.0+incompatible
       replaces: github.com/samalba/dockerclient=github.com/Dentrax/dockerclient@v0.1.0 github.com/dgrijalva/jwt-go=github.com/golang-jwt/jwt/v4@v4.4.2
 
+  - runs: |
+      sed -i 's|interim/master.xxxx|v5.3.2|g' controller/version.go
+
   - uses: go/build
     with:
       modroot: controller
-      ldflags: "-X google.golang.org/protobuf/reflect/protoregistry.conflictPolicy=warn"
+      ldflags: "-s -w -X google.golang.org/protobuf/reflect/protoregistry.conflictPolicy=warn"
       packages: .
       output: controller
 
+  - runs: |
+      # Create Folders
+      mkdir -p ${{targets.contextdir}}/etc
+      mkdir -p ${{targets.contextdir}}/usr/local/bin
+      mkdir -p ${{targets.contextdir}}/etc/neuvector/templates
+      mkdir -p ${{targets.contextdir}}/etc/tmp
+
+      # Scripts
+      cp scripts/sysctl.conf ${{targets.contextdir}}/etc/
+      cp scripts/teardown.sh ${{targets.contextdir}}/usr/local/bin/
+      cp scripts/runtime-gdb.py ${{targets.contextdir}}/usr/local/bin/
+
+      # Template
+      cp templates/podTemplate.json ${{targets.contextdir}}/etc/neuvector/templates/podTemplate.json
+      touch ${{targets.contextdir}}/usr/local/bin/.nvcontainer
+
+      # symlink
+      ln -sf /usr/bin/controller ${{targets.contextdir}}/usr/local/bin/controller
+      ln -sf /usr/bin/opa ${{targets.contextdir}}/usr/local/bin/opa
+      ln -sf /bin/consul ${{targets.contextdir}}/usr/local/bin/consul
+
+      # Retrieve Certs - with changes to openssl I can't generate them anymore.
+      mkdir -p ${{targets.contextdir}}/etc/neuvector/certs
+      wget https://raw.githubusercontent.com/neuvector/manifests/main/build/share/etc/neuvector/certs/ssl-cert.key -P ${{targets.contextdir}}/etc/neuvector/certs/
+      wget https://raw.githubusercontent.com/neuvector/manifests/main/build/share/etc/neuvector/certs/ssl-cert.pem -P ${{targets.contextdir}}/etc/neuvector/certs/
```

Comment:
None of our packages may ship private keys, or certs like these. Because that's insecure, and trips up security scanners.

Please explain the purpose of this key and cert, and why would a private key be the same for all installs and usage of this package? (meaning any install on any system can impersonate any other instance.....)

Shouldn't each deployment of this software deploy with its own self-managed certs?

Reply:
The openssl.cnf with stock CA settings was removed in https://github.com/wolfi-dev/os/pull/18053 precisely to prevent people from generating unmanaged certificates (i.e. certs without private key management, revocation, transparency logs, revocation urls, etc)

Reply:
I would expect a helm chart or vault to provide certificates; or use something like snakeoil; or generate certificates during image or package testing (see for example docker).

Reply:
sorry not docker, but mongodb tests.